### PR TITLE
Updates wasm middleware documentation with compilation instructions

### PR DESCRIPTION
**Pull request by codefromthecrypt**
**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This adds more background to the wasm docs including how to use it.
Before there were no instructions wrt how to generate the wasm used.

## Issue reference

https://github.com/dapr/components-contrib/pull/1833

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 1]
```diff
@@ -32,15 +40,15 @@ spec:
...
 
-| Field | Details | Example |
-|-------|---------|---------|
-| path | A relative or absolute path to the WASM module | "./hello.wasm" |
-| runtime | The WASM runtime of your WASM binary. Only `wazero` is supported | ["wazero"](https://github.com/tetratelabs/wazero) |
-
+| Field    | Details                                        | Example        |
+|----------|------------------------------------------------|----------------|
+| path     | A relative or absolute path to the Wasm binary | "./hello.wasm" |
+| poolSize | The instance count of the Wasm binary          | 10             |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: Why did you remove the `runtime` property from this table? Was this unintentional?

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: Also can you add the `poolsize` property into the YAML example.

Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: Need to say whether `poolsize is optional or required`. And describe a little more on what this does. Does it instantiate this number of Wasm binary images?

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 2]
```diff
@@ -54,8 +62,42 @@ spec:
...
+// rewrite returns a new URI if necessary.
+func rewrite(requestURI []byte) ([]byte, error) {
+	if string(requestURI) == "/v1.0/hi" {
+		return []byte("/v1.0/hello"), nil
+	}
+	return requestURI, nil
+}
+```
+
+If using TinyGo, compile like so and set the `path` attribute to the output:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
If using TinyGo, compile as shown below and set the `path` attribute to the output:
```
Also There is no `path` attribute below in the example

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md (hunk 1 above)
Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: > Why did you remove the runtime property from this table? Was this unintentional?

This was intentional because the only other runtimes that could work are build flags, not runtime flags, as they would require CGO (and are build flags in wapc-go). The first impl only supports one runtime, wazero as there's no plan to optimizing towards CGO per the below thread.

https://github.com/dapr/components-contrib/pull/1833#issuecomment-1186051640

> Also can you add the poolsize property into the YAML example.

sure can do

Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: > Need to say whether poolsize is optional or required. And describe a little more on what this does. Does it instantiate this number of Wasm binary images?

I will add text that it is optional. WRT how to describe it, the pool size is the amount of instances of the binary, specified by path. instance and binary is to wasm as container and image is to docker. Advice on rewording?

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md (hunk 2 above)
Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: > Also There is no path attribute below in the example

I'll reword to say "spec metadata field named path" unless you can figure a way to be unambiguous in a less chatty way.

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md (hunk 1 above)
Comment (codefromthecrypt) [reply daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]: I pitched a rewording. Let me know if it needs more clarity

**Review by msfussell — APPROVED** [review]
LGTM

**Review by greenie-msft — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
ps wazero started hosting a file that elaborates more on TinyGo, and I don't mind updating here if desired to point to it https://wazero.io/languages/tinygo/
